controllers/rhmi/rhmi_controller.go

```diff
@@ -1367,6 +1384,108 @@ func (r *RHMIReconciler) createInstallationCR(ctx context.Context, serverClient
 	return installation, nil
 }
 
+func (r *RHMIReconciler) composeAndSetAlertMetric(installation *rhmiv1alpha1.RHMI, rc *rest.Config, configManager *config.Manager) error {
+	var alerts []resources.AlertMetric
+
+	alertingNamespaces, err := r.getAlertingNamespace(installation, configManager)
+	if err != nil {
+		return fmt.Errorf("getting alerting namespace failed: %w", err)
+	}
+
+	observability, err := configManager.ReadObservability()
+	if err != nil {
+		return fmt.Errorf("getting observability configuration failed: %w", err)
+	}
+
+	for namespace, _ := range alertingNamespaces {
+		if namespace == observability.GetNamespace() {
+			alerts, err = r.composeAlertMetric("prometheus", namespace, rc)
+			if err != nil {
+				return fmt.Errorf("composing alert metric failed: %w", err)
+			}
+			metrics.SetRHOAMAlerts(alerts)
+		}
+	}
+
+	return nil
+}
+
+func (r *RHMIReconciler) composeAlertMetric(route string, namespace string, rc *rest.Config) ([]resources.AlertMetric, error) {
+	var alerts resources.AlertMetrics
+	endpoint := "/api/v1/alerts"
+
+	url, err := r.getURLFromRoute(route, namespace, rc)
+	if err != nil {
+		return alerts.Alerts, fmt.Errorf("error getting route : %w", err)
+	}
+	url = url + endpoint
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return alerts.Alerts, fmt.Errorf("error on request : %w", err)
+	}
+	var bearer = "Bearer " + r.restConfig.BearerToken
+	req.Header.Add("Authorization", bearer)
+
+	client := &http.Client{}
+	client.Timeout = time.Second * 10
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return alerts.Alerts, fmt.Errorf("error on response : %w", err)
+	}
+
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+			log.Error("reader close failure", err)
+		}
+	}(resp.Body)
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return alerts.Alerts, fmt.Errorf("unable to read body : %w", err)
+	}
+
+	var alertResp struct {
+		Status string `json:"status"`
+		Data   struct {
+			Alerts []struct {
+				Labels models.LabelSet `json:"labels"`
+				State  string          `json:"state"`
+			} `json:"alerts"`
+		} `json:"data"`
+	}
+
+	err = json.Unmarshal([]byte(body), &alertResp)
+	if err != nil {
+		return alerts.Alerts, fmt.Errorf("failed to unmarshal json: %w", err)
+	}
+
+	alerts = formatAlerts(alertResp.Data.Alerts)
+
+	return alerts.Alerts, nil
+}
+
+func (r *RHMIReconciler) setRHOAMClusterMetric() error {
+
+	infra, err := resources.GetClusterInfrastructure(context.TODO(), r.Client)
+	if err != nil {
+		return fmt.Errorf("error getting cluster infrastructure information: %w", err)
+	}
+
+	clusterType, err := resources.GetClusterType(infra)
+	if err != nil {
+		if clusterType == "Unknown" {
```

Comment:
This was intentional. Thinking about how these metrics will be used and the information it is given. There will be a dashboard given the count of the different clusters that rhoam is installed on (OSD, ROMA, GCP). While GCP is currently not supported at some point it will and I would think there is a high chance the `GetClusterType` function will be missed in the updates. If it is seems the SRE dashboards will still count the cluster but as a `Unknown` type. In my view it would be a far better experience to a `Unknown` type over a blank field ` `.

Read you comment for a second time. Would you still set the `metrics.SetRHOAMCluster` to an Unknown type?

Reply:
How can we reduce the chance of the function not getting updated? do we have enough information about what is shown in the infra cr to handle it now?

Reply:
From looking at the docs on the infra cr I didn't see any values that they could be. For the AWS values I checked actual installations for the values. 

To reduce the chance of the function not getting updated, I don't like my idea. We could put in stubs for the installations types (GCP, ...) that would crash the operator if that installation type was detected. While this would ensure the function gets updated, it seem wrong.